### PR TITLE
Add Airflow hook

### LINKED ISSUE #2266: Support Airflow Dag Triggering Hook
Part of #2231 
Add an hook that would trigger an Airflow DAG.
The hook should accept params like Airflow URL, DAG id and more. It should simple trigger the DAG and not wait for it to complete.

It's a useful use-case for `post-*` hooks, where you want to trigger some flow once the data is available and validated.

### PR DESCRIPTION
**Pull request by itaiad200**
closes #2266

### CODE REVIEW

**Review by talSofer — APPROVED** [review]
Very cool! 
Thanks for opening https://github.com/treeverse/lakeFS/issues/2282

File: pkg/actions/service_test.go [hunk 1]
```diff
@@ -38,57 +40,86 @@ func TestServiceRun(t *testing.T) {
...
+
+			username, pass, ok := r.BasicAuth()
+			require.True(t, ok)
+			require.Equal(t, "some_username", username)
+			require.Equal(t, "some_password", pass)
+			rawEvent, ok := req.Conf["lakeFS_event"]
+			require.True(t, ok, "missing lakeFS event")
+
+			var event actions.EventInfo
+			rawBytes, err := base64.StdEncoding.DecodeString(rawEvent.(string))
```
Comment (nopcoder) [inline pkg/actions/service_test.go:115]: encode json.RawMessage will prevent decode base64 and the actual implementation can parse the data as part of the original message

**Review by nopcoder — CHANGES_REQUESTED** [review]
minor comments

File: pkg/actions/service_test.go (hunk 1 above)
Comment (itaiad200) [reply pkg/actions/service_test.go:115]: Done, thanks!

File: pkg/actions/service_test.go [hunk 2]
```diff
@@ -38,57 +40,86 @@ func TestServiceRun(t *testing.T) {
...
+	const airflowHookID = "airflow_hook_id"
 	hookResponse := "OK"
 
-	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		defer func() { _ = r.Body.Close() }()
-		data, err := ioutil.ReadAll(r.Body)
-		if err != nil {
-			t.Error("Failed to read webhook post data", err)
-			return
+	checkEvent := func(event actions.EventInfo, hookID string) {
```
[the remark this answers is not in the document]
Comment (itaiad200) [reply pkg/actions/service_test.go:47]: Done.

File: pkg/actions/airflow.go [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+	// JSON object describing additional configuration parameters.
+	Conf map[string]interface{} `json:"conf,omitempty"`
+}
+
+func (a *Airflow) Run(ctx context.Context, record graveler.HookRecord, writer *HookOutputWriter) error {
+	eventData, err := marshalEventInformation(a.ActionName, a.ID, record)
+	if err != nil {
+		return err
+	}
+	a.AdditionalConf["lakeFS_event"] = eventData
```
[the remark this answers is not in the document]
Comment (itaiad200) [reply pkg/actions/airflow.go:89]: What do you mean?

File: pkg/actions/airflow.go [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+	airflowHook.Username, err = h.Properties.getRequiredString(airflowUsernamePropertyKey)
+	if err != nil {
+		return nil, fmt.Errorf("airflow hook username property: %w", err)
+	}
+	airflowHook.Password, err = h.Properties.getRequiredString(airflowPasswordPropertyKey)
+	if err != nil {
+		return nil, fmt.Errorf("airflow hook password property: %w", err)
+	}
+
+	conf, ok := h.Properties[airflowConf]
```
[the remark this answers is not in the document]
Comment (itaiad200) [reply pkg/actions/airflow.go:65]: Prefer to extract it when a second usage is needed, which I don't think is likely.

File: pkg/actions/webhook.go [hunk 5]
```diff
@@ -102,15 +103,28 @@ func (w *Webhook) Run(ctx context.Context, record graveler.HookRecord, writer *H
...
+	client := &http.Client{
+		Timeout: timeout,
+	}
 	start := time.Now()
 
-	buf := bytes.NewBufferString("Webhook request:\n")
+	var err error
+	buf := bytes.NewBufferString("Request:\n")
 	defer func() {
 		err2 := writer.OutputWrite(ctx, buf, int64(buf.Len()))
```
[the remark this answers is not in the document]
Comment (itaiad200) [reply pkg/actions/webhook.go:115]: Nice catch!

**Review by itaiad200 — COMMENTED** [review]
Thanks for the review! Fixed most of the comments, PTAL..

File: pkg/actions/airflow.go [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+	airflowHook.Password, err = h.Properties.getRequiredProperty(airflowPasswordPropertyKey)
+	if err != nil {
+		return nil, fmt.Errorf("airflow hook password property: %w", err)
+	}
+
+	conf, ok := h.Properties[airflowConf]
+	if ok {
+		airflowHook.DAGConf, ok = conf.(Properties)
+		if !ok {
+			return nil, fmt.Errorf("airflow hook conf is not of type 'map[string]interface{}': %w", errWrongValueType)
```
Comment (nopcoder) [inline pkg/actions/airflow.go:69]: Properties

File: pkg/actions/airflow.go [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+		}
+	}
+
+	return &airflowHook, nil
+}
+
+type DagRunReq struct {
+	// Run ID. This together with DAG_ID are a unique key.
+	DagRunID *string `json:"dag_run_id,omitempty"`
```
Comment (nopcoder) [inline pkg/actions/airflow.go:78]: missed this one - can you remove the pointer?

**Review by nopcoder — APPROVED** [review]

**Pull request merged**